sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateFormatter.scala

```
@@ -40,7 +43,23 @@ class Iso8601DateFormatter(
   extends DateFormatter with DateTimeFormatterHelper {
 
   @transient
-  private lazy val formatter = getOrCreateFormatter(pattern, locale)
+  private lazy val formatter = {
```

Shall we remove this lazy to let it fail fast in the parse phase? @cloud-fan

---

yea

---

Hmm, this one and the other are transient, so the lazy keyword is required.